content/5_Plat/2DRQ.mdx

```diff
@@ -322,93 +344,89 @@ And you might use it like so:
 
 <LanguageSection>
 <CPPSection>
+
 ```cpp
-#include <algorithm>
-#include <cassert>
-#include <iostream>
-#include <vector>
+#include <bits/stdc++.h>
 using namespace std;
 using ll = long long;
 
-// BeginCodeSnip{BIT2D}
-// index of largest value <= x in v (sorted)
-// if v = [1, 2, 4], ind(v, 3) would return 1
-int ind(vector<int> v, int x) {
-	return upper_bound(v.begin(), v.end(), x) - v.begin() - 1;
-}
-
-class BIT2D {
+/**
+ * Offline 2D Fenwick Tree implementation.
+ * Note that n needs to be of a reasonable size, and
+ * all the updates/queries inputs are zero indexed.
+ */
+template <typename T> class OfflineBIT2D {
   private:
-	int n;  // max x-coordinate
-	vector<vector<int>> vals, bit;
+	const int n;
+	vector<vector<int>> vals;
+	vector<vector<T>> bit;
+
+	int ind(const vector<int> &v, int x) {
+		return upper_bound(begin(v), end(v), x) - begin(v) - 1;
+	}
 
   public:
-	BIT2D(int n, vector<pair<int, int>> &todo) : n(n), vals(n + 1), bit(n + 1) {
-		// sort points by y-coordinate
-		sort(todo.begin(), todo.end(),
-		     [](pair<int, int> a, pair<int, int> b) { return a.second < b.second; });
-		// ensures vals and bit are 1-indexed
+	OfflineBIT2D(int _n, vector<array<int, 2>> &todo) : n(_n), vals(n + 1), bit(n + 1) {
```

Comment:
same thing abt the underscore

Reply:
oh shoot, i forgot to change that part LOl